pkgs/development/compilers/llvm/common.nix

```diff
@@ -2,7 +2,9 @@
 
 rec {
   llvmBackend = platform:
-    if platform.parsed.cpu.family == "x86" then
+    if builtins.typeOf platform == "string" then
```

Comment:
Fixed. But I still do not like this.

Reply:
We could stringify our default targets before passing them to the list.

Reply:
I think the ability to pass platform attribute sets to `enabledTargets` is useful and convenient. We might want to move the llvm target name to `lib/platforms.nix`; it already contains gcc keys. Then we might simplify the function to
```nix
 llvmBackend = platform:
    if builtins.typeOf platform == "string" then
      platform
    else if stdenv.lib.hasAttrByPath ["llvm" "targetName" ] then
      platform.llvm.targetName
    else
      throw "Unsupported system";
```

Reply:
Oh that's a good point. CC @matthewbauer.